src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -1,242 +1,54 @@
-                { ReleaseBranchRegex, 30000 },
-                { FeatureBranchRegex, 30000 },
-                { PullRequestRegex, 30000 },
-                { SupportBranchRegex, 55000 },
-                { MasterBranchRegex, 55000 }
-            };
-
-        private const IncrementStrategy DefaultIncrementStrategy = IncrementStrategy.Inherit;
+        public ConfigurationProvider(IFileSystem fileSystem, ILog log, IConfigFileLocator configFileLocator, IGitPreparer gitPreparer, IConfigInitWizard configInitWizard)
+        {
+            this.fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            this.log = log ?? throw new ArgumentNullException(nameof(log));
+            this.configFileLocator = configFileLocator ?? throw new ArgumentNullException(nameof(configFileLocator));
+            this.gitPreparer = gitPreparer ?? throw new ArgumentNullException(nameof(gitPreparer));
+            this.configInitWizard = configInitWizard ?? throw new ArgumentNullException(nameof(this.configInitWizard));
+        }
 
-        public static Config Provide(GitPreparer gitPreparer, IConfigFileLocator configFileLocator, bool applyDefaults = true, Config overrideConfig = null)
+        public Config Provide(bool applyDefaults = true, Config overrideConfig = null)
         {
             var workingDirectory = gitPreparer.WorkingDirectory;
             var projectRootDirectory = gitPreparer.GetProjectRootDirectory();
 
             if (configFileLocator.HasConfigFileAt(workingDirectory))
             {
-                return Provide(workingDirectory, configFileLocator, applyDefaults, overrideConfig);
+                return Provide(workingDirectory, applyDefaults, overrideConfig);
             }
 
-            return Provide(projectRootDirectory, configFileLocator, applyDefaults, overrideConfig);
+            return Provide(projectRootDirectory, applyDefaults, overrideConfig);
         }
 
-        public static Config Provide(string workingDirectory, IConfigFileLocator configFileLocator, bool applyDefaults = true, Config overrideConfig = null)
+        public Config Provide(string workingDirectory, bool applyDefaults = true, Config overrideConfig = null)
         {
             var readConfig = configFileLocator.ReadConfig(workingDirectory);
-            VerifyConfiguration(readConfig);
+            ConfigurationUtils.VerifyConfiguration(readConfig);
 
-            if (applyDefaults)
-                ApplyDefaultsTo(readConfig);
-            if (null != overrideConfig)
-                ApplyOverridesTo(readConfig, overrideConfig);
+            if (applyDefaults) ConfigurationUtils.ApplyDefaultsTo(readConfig);
+            if (null != overrideConfig) ConfigurationUtils.ApplyOverridesTo(readConfig, overrideConfig);
             return readConfig;
         }
 
-        private static void VerifyConfiguration(Config readConfig)
+        public string GetEffectiveConfigAsString(string workingDirectory)
```

Comment:
Can't this just be a `ToString()` override on the `Config` class?

Reply:
Created the `ToString()` override into Config class